<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="en">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.4" rel="stylesheet" type="text/css">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=5.1.4">


  <link rel="mask-icon" href="/images/logo.svg?v=5.1.4" color="#222">





  <meta name="keywords" content="Hexo, NexT">










<meta property="og:type" content="website">
<meta property="og:title" content="Juan-NF&#39;s blog">
<meta property="og:url" content="http://yoursite.com/index.html">
<meta property="og:site_name" content="Juan-NF&#39;s blog">
<meta property="og:locale" content="en">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Juan-NF&#39;s blog">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Pisces',
    version: '5.1.4',
    sidebar: {"position":"left","display":"post","offset":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    duoshuo: {
      userId: '0',
      author: 'Author'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://yoursite.com/">





  <title>Juan-NF's blog</title>
  








</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="en">

  
  
    
  

  <div class="container sidebar-position-left 
  page-home">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Juan-NF's blog</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle">Be curious</p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            Home
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            Archives
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            
  <section id="posts" class="posts-expand">
    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2019/08/31/生存分析/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Juan-NF">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.gif">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Juan-NF's blog">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2019/08/31/生存分析/" itemprop="url">生存分析</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">Posted on</span>
              
              <time title="Post created" itemprop="dateCreated datePublished" datetime="2019-08-31T19:44:27+08:00">
                2019-08-31
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h4 id="1-TCGA数据库-patient-barcode–那些需要了解的“暗号”"><a href="#1-TCGA数据库-patient-barcode–那些需要了解的“暗号”" class="headerlink" title="1.TCGA数据库 patient_barcode–那些需要了解的“暗号”"></a>1.TCGA数据库 patient_barcode–那些需要了解的“暗号”</h4><ul>
<li>在TCGA数据库中，相同的参与者可能既有肿瘤组织也有正常组织的数据，生存分析时需要将肿瘤数据选出来，进行下一步分析；如果是做差异分析，则可能可能是比较肿瘤和正常组织；</li>
<li>详细的可以看<a href="https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/" target="_blank" rel="noopener">概况</a>和<a href="https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables" target="_blank" rel="noopener">细节</a></li>
</ul>
<img src="https://raw.githubusercontent.com/Juan-NF/Figurebed/master/img/20190831194723.png">

<ul>
<li>最近用得较多的是，通过Sample中的数字部分来区分<strong>tumor</strong>(<strong>&lt;10</strong>)和<strong>normal</strong>(<strong>&gt;=10</strong>)</li>
</ul>
<table>
<thead>
<tr>
<th>编码</th>
<th>代表内容</th>
<th>缩写</th>
</tr>
</thead>
<tbody><tr>
<td>1</td>
<td>Primary Solid Tumor</td>
<td>TP</td>
</tr>
<tr>
<td>2</td>
<td>Recurrent Solid Tumor</td>
<td>TR</td>
</tr>
<tr>
<td>3</td>
<td>Primary Blood Derived Cancer - Peripheral Blood</td>
<td>TB</td>
</tr>
<tr>
<td>4</td>
<td>Recurrent Blood Derived Cancer - Bone Marrow</td>
<td>TRBM</td>
</tr>
<tr>
<td>5</td>
<td>Additional - New Primary</td>
<td>TAP</td>
</tr>
<tr>
<td>6</td>
<td>Metastatic</td>
<td>TM</td>
</tr>
<tr>
<td>7</td>
<td>Additional Metastatic</td>
<td>TAM</td>
</tr>
<tr>
<td>8</td>
<td>Human Tumor Original Cells</td>
<td>THOC</td>
</tr>
<tr>
<td>9</td>
<td>Primary Blood Derived Cancer - Bone Marrow</td>
<td>TBM</td>
</tr>
<tr>
<td>10</td>
<td>Blood Derived Normal</td>
<td>NB</td>
</tr>
<tr>
<td>11</td>
<td>Solid Tissue Normal</td>
<td>NT</td>
</tr>
<tr>
<td>12</td>
<td>Buccal Cell Normal</td>
<td>NBC</td>
</tr>
<tr>
<td>13</td>
<td>EBV Immortalized Normal</td>
<td>NEBV</td>
</tr>
<tr>
<td>14</td>
<td>Bone Marrow Normal</td>
<td>NBM</td>
</tr>
<tr>
<td>15</td>
<td>sample type 15</td>
<td>15SH</td>
</tr>
<tr>
<td>16</td>
<td>sample type 16</td>
<td>16SH</td>
</tr>
<tr>
<td>20</td>
<td>Control Analyte</td>
<td>CELLC</td>
</tr>
<tr>
<td>40</td>
<td>Recurrent Blood Derived Cancer - Peripheral Blood</td>
<td>TRB</td>
</tr>
<tr>
<td>50</td>
<td>Cell Lines</td>
<td>CELL</td>
</tr>
<tr>
<td>60</td>
<td>Primary Xenograft Tissue</td>
<td>XP</td>
</tr>
<tr>
<td>61</td>
<td>Cell Line Derived Xenograft Tissue</td>
<td>XCL</td>
</tr>
<tr>
<td>99</td>
<td>sample type 99</td>
<td>99SH</td>
</tr>
</tbody></table>
<h4 id="2-TCGA数据下载"><a href="#2-TCGA数据下载" class="headerlink" title="2.TCGA数据下载"></a>2.TCGA数据下载</h4><ul>
<li><p>数据下载</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">library(TCGAbiolinks)</span><br><span class="line">query &lt;- GDCquery(project = &quot;TCGA-BRCA&quot;,</span><br><span class="line">                  data.category = &quot;Transcriptome Profiling&quot;,</span><br><span class="line">                  data.type = &quot;Gene Expression Quantification&quot;, </span><br><span class="line">                  workflow.type = &quot;HTSeq - Counts&quot;,</span><br><span class="line">                  barcode = clin[1:500,1])</span><br><span class="line">GDCdownload(query)</span><br><span class="line">BRC_DATA1&lt;-GDCprepare(query,save=FALSE)</span><br><span class="line">library(SummarizedExperiment)</span><br><span class="line">BRC_DATA1_1 &lt;- assay(BRC_DATA1)</span><br></pre></td></tr></table></figure>
</li>
<li><p>workflow.type有多个选择，HTSeq - Counts，HTSeq - FPKM-UQ，HTSeq - FPKM等；</p>
</li>
<li><p>GDCquery设置好要下载的内容；</p>
</li>
<li><p>GDCdownload进行下载；如果已经下载过了，运行该步骤会有数据已经下载过的提示；</p>
</li>
<li><p>GDCprepare将下载的数据（文件夹套文件夹的格式）进行整理，得到一个表格；但这次最后得到的是一个SummarizedExperiment objects，用<code>SummarizedExperiment</code>中的assay将内容提取出来即可；</p>
</li>
<li><p>临床数据下载</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">library(TCGAbiolinks)</span><br><span class="line">clinic &lt;- GDCquery_clinic(project = &quot;TCGA-BRCA&quot;,</span><br><span class="line">                          type = &apos;Clinical&apos;)</span><br></pre></td></tr></table></figure>
</li>
<li><p>个性化的<code>TNBC</code>分型</p>
<blockquote>
<p>TCGA中TNBC的数据是基于Her2、ER、PR的IHC结果来进行划分的，相应的表型数据是这样获得的：内容整理自<a href="https://www.biostars.org/p/279048/" target="_blank" rel="noopener">TNBCsub</a></p>
</blockquote>
</li>
</ul>
<p>&emsp;1.点击链接[GDC Data]，选择界面右下角<strong>Legacy Archive</strong>(<a href="https://portal.gdc.cancer.gov/" target="_blank" rel="noopener">https://portal.gdc.cancer.gov/</a>)<br><img src="https://raw.githubusercontent.com/Juan-NF/Figurebed/master/img/20190831195145.png"><br>&emsp;2.Primary Site和Project选择；<br><img src="https://raw.githubusercontent.com/Juan-NF/Figurebed/master/img/20190831195241.png"><br>&emsp;3.Files中，Data Type选择Clinical data，Data Format选择Biotab，右侧文件选择<code>nationwidechildrens.org_clinical_patient_brca.txt</code>；<br><img src="https://raw.githubusercontent.com/Juan-NF/Figurebed/master/img/20190831195325.png"></p>
<h4 id="3-KMplot"><a href="#3-KMplot" class="headerlink" title="3.KMplot"></a>3.KMplot</h4><blockquote>
<p>a.看下这个图是用了什么数据?<br>b.看下这些数据是怎么来的?<br>c.如何区分是否有差异？<strong>log-rank test</strong></p>
<ul>
<li>一系列代码</li>
</ul>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">library(survival)</span><br><span class="line">library(survminer)</span><br><span class="line">fit.surv1 &lt;-Surv(tumor_clin$time,as.numeric(tumor_clin$Status))</span><br><span class="line">fit&lt;-survfit(fit.surv1~group,data=voom_group)</span><br><span class="line">summary(fit)</span><br><span class="line">ggsurvplot(fit)</span><br></pre></td></tr></table></figure>

<p><code>tumor_clin$time</code>生存分析中的时间；<br><code>tumor_clin$Status</code>生存分析中的Alive-<strong>0</strong>和Dead-<strong>1；</strong><br><code>group</code>对应的数据中的分组信息的列的<strong>列名；</strong><br><code>summary（fit）</code>查看数据<br><img src="https://raw.githubusercontent.com/Juan-NF/Figurebed/master/img/20190831195454.png"></p>
<ul>
<li><p>a中提到的图，即是下图，其实KMplot主要的目的即是<strong>生存曲线可视化</strong></p>
<img src="https://raw.githubusercontent.com/Juan-NF/Figurebed/master/img/20190831195410.png"></li>
<li><p>演示survival的计算（计算了一组）：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">clin_group&lt;- tumor_clin[,c(&apos;Status&apos;,&apos;time&apos;)]</span><br><span class="line">group &lt;- voom_group[,1]</span><br><span class="line">#####取出对应分组的临床信息</span><br><span class="line">group_1 &lt;- clin_group[group==1,]</span><br><span class="line">group_2 &lt;- clin_group[group==2,]</span><br><span class="line">#####取出对应分组的对应Status（Alive|Death）信息</span><br><span class="line">group_1_A&lt;- group_1[group_1$Status==0,]</span><br><span class="line">group_1_D&lt;- group_1[group_1$Status==1,]</span><br><span class="line">group_2_A&lt;- group_2[group_2$Status==0,]</span><br><span class="line">group_2_D&lt;- group_2[group_2$Status==1,]</span><br><span class="line">#####对1组下的Status信息按照time进行排序</span><br><span class="line">A_1&lt;- group_1_A[order(group_1_A$time),]</span><br><span class="line">D_1&lt;- group_1_D[order(group_1_D$time),]</span><br><span class="line">######将相同time的event的计数进行加和</span><br><span class="line">D_1&lt;- data.frame(time = as.numeric(D_1$time[!duplicated(D_1$time)]),</span><br><span class="line">               event= as.numeric(by(D_1,D_1$time, function(x)&#123;length(x[,1])&#125;)))</span><br><span class="line">#####将相同time的censor（alive）的计数进行加和</span><br><span class="line">A_1&lt;- data.frame(time = as.numeric(A_1$time[!duplicated(A_1$time)]),</span><br><span class="line">                 censor= as.numeric(by(A_1,A_1$time, function(x)&#123;length(x[,1])&#125;)))</span><br><span class="line">#####这一组的总人数-event（death）-censor（alive|依据时间排序后，`此death对应时间`之前的alive）</span><br><span class="line">y&lt;- function(x)&#123;nrow(group_1)-sum(D_1$event[D_1$time&lt;x[1]])-sum(A_1$censor[A_1$time&lt;x[1]])&#125;</span><br><span class="line">D_1$n.risk&lt;- apply(D_1,1,y)</span><br><span class="line">#####每一个时间点的survival=（risk（number of alive）-event）/risk（number of alive）</span><br><span class="line">D_1$step.survival &lt;- apply(D_1,1,function(x)&#123;(x[3]-x[2])/x[3]&#125;)</span><br><span class="line">#####截止到此时间点的survival=此时间点之前的survival*此时间点的survival</span><br><span class="line">D_1$surv &lt;- apply(D_1,1,function(x)&#123;prod(D_1$step.survival[D_1$time&lt;=x[1]])&#125;)</span><br></pre></td></tr></table></figure>

<ul>
<li>至此所求为1组的数据，2组数据同理；</li>
<li>KMplot涉及到的数据很少，Status、time和分组信息（依据处理或表达量等进行分组）等等，Status中一般Alive和Death用0,1代表或1,2代表（记住就行，别问为什么）；</li>
</ul>
</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">&gt; D_1</span><br><span class="line">   time event n.risk step.survival      surv</span><br><span class="line">1    26     1    519     0.9980732 0.9980732</span><br><span class="line">2   116     1    501     0.9980040 0.9960811</span><br><span class="line">3   158     1    500     0.9980000 0.9940889</span><br><span class="line">4   160     1    499     0.9979960 0.9920967</span><br><span class="line">5   172     1    496     0.9979839 0.9900965</span><br><span class="line">6   174     1    494     0.9979757 0.9880923</span><br><span class="line">7   302     1    480     0.9979167 0.9860338</span><br><span class="line">8   320     1    473     0.9978858 0.9839491</span><br><span class="line">9   322     1    471     0.9978769 0.9818601</span><br><span class="line">10  336     1    467     0.9978587 0.9797576</span><br><span class="line">11  348     1    463     0.9978402 0.9776415</span><br><span class="line">12  365     1    458     0.9978166 0.9755069</span><br><span class="line">13  377     1    447     0.9977629 0.9733245</span><br><span class="line">14  385     2    441     0.9954649 0.9689104</span><br><span class="line">15  426     1    423     0.9976359 0.9666198</span><br><span class="line">16  446     1    412     0.9975728 0.9642736</span><br><span class="line">17  524     1    372     0.9973118 0.9616815</span><br><span class="line">18  538     1    366     0.9972678 0.9590540</span><br><span class="line">19  558     1    355     0.9971831 0.9563524</span><br><span class="line">20  571     1    351     0.9971510 0.9536278</span><br><span class="line">21  584     1    344     0.9970930 0.9508556</span><br><span class="line">22  612     1    332     0.9969880 0.9479916</span><br><span class="line">23  614     1    331     0.9969789 0.9451275</span><br><span class="line">#####未展示所有，画图用的是time和surv这两列</span><br></pre></td></tr></table></figure>

<h4 id="4-log-rank检验"><a href="#4-log-rank检验" class="headerlink" title="4.log-rank检验"></a>4.log-rank检验</h4><ul>
<li><p>log-rank检验是比较生存曲线是否有差异的最常用的方法，非参数检验。</p>
</li>
<li><p>零假设是两组之间的生存率没有差异。</p>
</li>
<li><p>log-rank统计量大致满足为卡方分布。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">fit.surv1 &lt;-Surv(tumor_clin$time,as.numeric(tumor_clin$Status))</span><br><span class="line">log_rank_p &lt;- apply(tumor_voom, 1, function(values1)&#123;</span><br><span class="line">  group=ifelse(values1&gt;median(values1),2,1)</span><br><span class="line">  data.survdiff=survdiff(fit.surv1~group)</span><br><span class="line">  p.val = 1 - pchisq(data.survdiff$chisq, length(data.survdiff$n) - 1)</span><br><span class="line">  return( p.val)</span><br><span class="line">&#125;)</span><br></pre></td></tr></table></figure>
</li>
<li><p>这里是批量log-rank检验的代码；</p>
</li>
<li><p>用survdiff函数的<code>$chisq</code>、pchisq函数、<code>df=length(data.survdiff$n) - 1</code>对p值进行计算（上面有提到，大致满足卡方分布）；</p>
</li>
<li><p>survdiff的结果返回如下：<code>Oberved</code>是观察到的event的个数；<code>Expected</code>是理论的event的个数；<code>(O-E)^2/E</code>即直观反应到的计算结果；<code>V</code>用<code>data.survdiff$var</code>可以得到（还没细研究）；<code>(O-E)^2/V</code>即构建的log-rank的统计量，满足卡方分布的是它；</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">&gt; data.survdiff</span><br><span class="line">Call:</span><br><span class="line">survdiff(formula = fit.surv1 ~ group, data = voom_group)</span><br><span class="line"></span><br><span class="line">          N Observed Expected (O-E)^2/E (O-E)^2/V</span><br><span class="line">group=1 545       79     72.1     0.656      1.25</span><br><span class="line">group=2 545       73     79.9     0.592      1.25</span><br><span class="line"></span><br><span class="line"> Chisq= 1.3  on 1 degrees of freedom, p= 0.3</span><br></pre></td></tr></table></figure>

</li>
</ul>
<h4 id="5-coxph（Cox-proportional-hazards）"><a href="#5-coxph（Cox-proportional-hazards）" class="headerlink" title="5.coxph（Cox proportional hazards）"></a>5.coxph（Cox proportional hazards）</h4><ul>
<li><p>用于描述不同变量对于生存的影响； </p>
</li>
<li><p>该方法不对“生存模型”做出假设，假设变量对生存的影响随时间变化是恒定的，并且在一个尺度中具有累加效应，因此不是真正的非参数，为半参数；生存曲线可视化无交叉表示满足PH设定；</p>
</li>
<li><p>单因素cox批量的代码：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">fit.surv1 &lt;-Surv(tumor_clin$time,as.numeric(tumor_clin$Status))</span><br><span class="line">cox_p &lt;- apply(tumor_voom, 1, function(values1)&#123;</span><br><span class="line">  group=ifelse(values1&gt;median(values1),2,1)</span><br><span class="line">  fit &lt;- coxph(fit.surv1~group)</span><br><span class="line">  fit1 &lt;- exp(fit$coefficients)</span><br><span class="line">  p.val &lt;- summary(fit)$logtest[&quot;pvalue&quot;]</span><br><span class="line">  result &lt;- list(fit1,p.val)</span><br><span class="line">  return(result)</span><br><span class="line">&#125;)</span><br></pre></td></tr></table></figure>
</li>
<li><p>分组时，将组别用数字2和1来代替，比如，根据表达量高低进行分组，看某基因高表达是风险增加还是降低；所给出的HR即使<code>组2</code>相对<code>组1</code>的值；</p>
</li>
<li><p>结果格式如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">&gt; summary(coxph)</span><br><span class="line">Call:</span><br><span class="line">coxph(formula = fit.surv1 ~ group, data = voom_group)</span><br><span class="line"></span><br><span class="line">  n= 1090, number of events= 152 </span><br><span class="line"></span><br><span class="line">         coef exp(coef) se(coef)      z Pr(&gt;|z|)</span><br><span class="line">group -0.1818    0.8338   0.1627 -1.117    0.264</span><br><span class="line"></span><br><span class="line">      exp(coef) exp(-coef) lower .95 upper .95</span><br><span class="line">group    0.8338      1.199    0.6061     1.147</span><br><span class="line"></span><br><span class="line">Concordance= 0.546  (se = 0.024 )</span><br><span class="line">Likelihood ratio test= 1.25  on 1 df,   p=0.3</span><br><span class="line">Wald test            = 1.25  on 1 df,   p=0.3</span><br><span class="line">Score (logrank) test = 1.25  on 1 df,   p=0.3</span><br></pre></td></tr></table></figure>
</li>
<li><p><code>exp（coef）</code> 即HR，此处为0.8338；</p>
<blockquote>
<p>HR = 1: 无风险<br>HR &lt; 1: 风险降低<br>HR &gt; 1: 风险增加</p>
</blockquote>
</li>
<li><p><code>lower .95</code>和<code>upper .95</code>为HR的<strong>95%CI</strong>的下限和上限；可以通过<code>$conf.int[3]</code>和<code>$conf.int[4]</code>调出来；</p>
</li>
<li><p>Likelihood ratio test、wald test和Score（logrank）test的检验结果均有展示；</p>
</li>
</ul>
<p>【参考内容】<br>1.<a href="https://www.jianshu.com/p/33bdb5ef7689" target="_blank" rel="noopener">各种格式TCGA数据下载</a><br>2.<a href="http://www.sthda.com/english/wiki/cox-proportional-hazards-model" target="_blank" rel="noopener">cox-survival</a><br>3.<a href="https://blog.csdn.net/xiaohukun/article/details/77936022" target="_blank" rel="noopener">survival rate计算</a></p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2019/08/31/alignment/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Juan-NF">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.gif">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Juan-NF's blog">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2019/08/31/alignment/" itemprop="url">alignment</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">Posted on</span>
              
              <time title="Post created" itemprop="dateCreated datePublished" datetime="2019-08-31T15:58:12+08:00">
                2019-08-31
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h4 id="fastq1和fastq2特定的cluster的序列"><a href="#fastq1和fastq2特定的cluster的序列" class="headerlink" title="fastq1和fastq2特定的cluster的序列"></a>fastq1和fastq2特定的cluster的序列</h4><blockquote>
<ul>
<li>@SRR8980088.3 fastq的identifier</li>
<li>基于此可以找到下机数据中该编号对应的序列；</li>
</ul>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">a=/trainee/vip13/test</span><br><span class="line">####查看对应的序列</span><br><span class="line">zcat /trainee/vip13/test/SRR8980088_2_val_2.fq.gz|grep -A1 &apos;SRR8980088.3&apos;|head -n2|grep -v @</span><br><span class="line">zcat /trainee/vip13/test/SRR8980088_1_val_1.fq.gz|grep -A1 &apos;SRR8980088.3&apos;|head -n2|grep -v @</span><br></pre></td></tr></table></figure>

<p><img src="https://upload-images.jianshu.io/upload_images/13774856-c8434e6bcd53c0e9.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="fastq1"></p>
<p><img src="https://upload-images.jianshu.io/upload_images/13774856-1dc0e8b1209b322e.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="fastq2"></p>
<h4 id="fastq在SAM-BAM文件中的展示"><a href="#fastq在SAM-BAM文件中的展示" class="headerlink" title="fastq在SAM|BAM文件中的展示"></a>fastq在SAM|BAM文件中的展示</h4><blockquote>
<ul>
<li>基于此，可以在SAM/BAM文件中找到，其比对到fasta上的序列</li>
<li>在这里的体现是：fastq1与fa方向一致，fastq2与fa反向互补；</li>
</ul>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat /trainee/vip13/test/SRR8980088.sam|grep SRR8980088.3|head -n2|cut -f1,10</span><br></pre></td></tr></table></figure>

<p><img src="https://upload-images.jianshu.io/upload_images/13774856-d3f36ba1bcb5d0a8.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="SAM/BAM"></p>
<h4 id="Fastq中的read在fasta中的位置-SAM-BAM"><a href="#Fastq中的read在fasta中的位置-SAM-BAM" class="headerlink" title="Fastq中的read在fasta中的位置-SAM/BAM"></a>Fastq中的read在fasta中的位置-SAM/BAM</h4><blockquote>
<ul>
<li>上述identifier由比对结果看，位于chr1；</li>
<li>从fasta中将chr1序列提取出来，截取出其对应的碱基位置，进行查看；</li>
<li>依据，SAM中对于pos的解释，为比对上的reference左端的位置，所以，最后截取的reference的位置，为 <code>左（小）~（左（大）+读长-1）</code>，这也就是<code>echo $((110954876+149))</code>的来由；</li>
<li><code>grep -i</code>所跟序列为SAM|BAM中所对应的比对序列；</li>
</ul>
</blockquote>
<p><img src="https://upload-images.jianshu.io/upload_images/13774856-7d79329f2226c0f2.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sam/bam/pos"><br><img src="https://upload-images.jianshu.io/upload_images/13774856-b2549380c211e026.jpg?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="lib_PEseq.jpg"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">##从fa文件将chr1序列提取出来</span><br><span class="line">zcat /trainee/vip13/test/hg38.fa.gz |sed -n &apos;/chr1$/,/chr10$/p&apos; &gt; test.txt</span><br><span class="line">###需要精确确定位置，所以，chr的字样需要去除</span><br><span class="line">cat test.txt |grep -v &apos;&gt;chr&apos; &gt; chr1.txt</span><br><span class="line">echo $((110954876+149))</span><br><span class="line">###相应的，换行符等也要去掉</span><br><span class="line">cat /trainee/vip13/test/chr1.txt |tr -d &apos;\n&apos;|cut -c 110954736-110955025|grep -i</span><br></pre></td></tr></table></figure>

<p><img src="https://upload-images.jianshu.io/upload_images/13774856-70dab7565e788828.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sam_fq1"><br><img src="https://upload-images.jianshu.io/upload_images/13774856-6da2039cb89112f7.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sam_fq2"></p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
  </section>

  


          </div>
          


          

        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      

      <section class="site-overview-wrap sidebar-panel sidebar-panel-active">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <p class="site-author-name" itemprop="name">Juan-NF</p>
              <p class="site-description motion-element" itemprop="description"></p>
          </div>

          <nav class="site-state motion-element">

            
              <div class="site-state-item site-state-posts">
              
                <a href="/archives/">
              
                  <span class="site-state-item-count">2</span>
                  <span class="site-state-item-name">posts</span>
                </a>
              </div>
            

            

            

          </nav>

          

          

          
          

          
          

          

        </div>
      </section>

      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2019</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Juan-NF</span>

  
</div>


  <div class="powered-by">Powered by <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a></div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">Theme &mdash; <a class="theme-link" target="_blank" href="https://github.com/iissnan/hexo-theme-next">NexT.Pisces</a> v5.1.4</div>




        







        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>
  

  
  
    <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.4"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.4"></script>



  

  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.4"></script>



  


  




	





  





  












  





  

  

  

  
  

  

  

  

</body>
</html>
